### PR TITLE
Datepicker Dot Markers

### PR DESCRIPTION
**Pull request by isacoder**
### Description

This is a new Feature request for the Datepicker

`
A new date picker feature where a list of dates or date ranges can be provided which will then place a dot on the date picker to signify, for example, that there is content available to view within that range.
`

 ### Considerations on Implementation
Moved examples data to new data helper file for DatePicker examples.
I decided to add go with the dates array for content.

 ### Reviewing/Testing steps
Test in Storybook

Filters -> Molecules -> DatePicker

Screenshot 2024-12-05 at 13 48 13

Filters -> Organism -> FilterBar

Screenshot 2024-12-05 at 13 48 43

### CODE REVIEW

File: packages/ui-lib/src/Filters/molecules/DatePicker.tsx [hunk 1]
```diff
@@ -195,6 +195,29 @@ const CalHCell = styled(CalCell)`
...
 `;
 
+const ContentDot = styled.div<{ hasContent: boolean, state?: CellStates, isToday?: boolean, }>`
+  position: absolute;
+  left: 18px;
+  bottom: 5px;
+  width: 4px;
+  height: 4px;
+  border-radius: 50%;
+  background-color: var(--info-10);
```
Comment (atomworks) [inline packages/ui-lib/src/Filters/molecules/DatePicker.tsx:205]: Can we update this to `--primary-11` please.

File: packages/ui-lib/src/Filters/molecules/DatePicker.tsx [hunk 2]
```diff
@@ -195,6 +195,29 @@ const CalHCell = styled(CalCell)`
...
+
+  ${({ state }) => (state === 'single' || state === 'start' || state === 'end' || state === 'inside') && css`
+    background-color: var(--white-12);`
```
Comment (atomworks) [inline packages/ui-lib/src/Filters/molecules/DatePicker.tsx:208]: For `inside` can we make this `--primary-12` please. The other states should remain `--white-12`, thank you.

**Review by atomworks — CHANGES_REQUESTED** [review]
Looking great... just three things, from small to big. Depending on your thoughts, the third might not need any work right now.

1. I'd like to revise the dot colors slightly - I'll leave the changes in the code comments.
2. While the dot is in the right place, the number itself needs to go up by 1px.
3. --The date array approach looks great though I'm concerned that it's a little messy to have required in this field again. I know we have it in the date range fields themselves but as I'd like to revise and remove that in the future when we can start to make breaking changes, I'd rather avoid it for now. What are your thoughts?--

Thank you!

**Comment by isacoder** [comment]
I have updated the colors, I will review with @atomworks  points 2 and 3 to understand better these requirements before updating the PR.

Screenshot 2024-12-06 at 15 31 52

**Review by atomworks — APPROVED** [review]
Looks ready to go, thank you!

**Pull request merged**